src/bin/main.rs

```diff
@@ -23,3 +52,176 @@ async fn main() -> Result<(), Error> {
+    let data_dir = args
+        .data_dir
+        .as_ref()
+        .cloned()
+        .unwrap_or_else(WalletPath::default_dir);
+
+    // create directories
+    fs::create_dir_all(&data_dir)?;
+
+    // set cache directory straight away
+    WalletPath::set_cache_dir(&data_dir)?;
+
+    // load configuration (or use default)
+    let mut cfg = Config::load(data_dir)?;
+
+    // merge static config with parsed args
+    cfg.merge(args);
+
+    // get command or default to interactive mode
+    let cmd = cmd.unwrap_or(Command::Interactive);
+
+    // request auth for wallet (if required)
+    let pwd = if cmd.uses_wallet() || cfg.wallet.file.is_some() {
+        prompt::request_auth("Please enter wallet password")
+    } else {
+        blake3::hash("".as_bytes())
+    };
+
+    // prepare wallet path
+    let wallet_path = match cfg.wallet.file {
+        Some(ref p) => WalletPath::from(p.with_extension("dat")),
+        None => {
+            let mut pb = PathBuf::new();
+            pb.push(&cfg.wallet.data_dir);
+            pb.push(&cfg.wallet.name);
+            pb.set_extension("dat");
+            WalletPath::from(pb)
+        }
+    };
+
+    // get our wallet ready
+    let mut wallet: Wallet<WalletFile> = match cmd {
+        Command::Create => {
+            // create a new randomly generated mnemonic phrase
+            let mnemonic =
+                Mnemonic::new(MnemonicType::Words12, Language::English);
+            // ask user for a password to secure the wallet
+            let pwd = prompt::create_password();
+            // skip phrase confirmation if explicitly
+            if !cfg.wallet.skip_recovery {
+                prompt::confirm_recovery_phrase(&mnemonic);
+            }
+            // create wallet
+            let mut w = Wallet::new(mnemonic)?;
+            w.save_to(WalletFile {
+                path: wallet_path,
+                pwd,
+            })?;
+            w
+        }
+        Command::Restore => {
+            // ask user for 12-word recovery phrase
+            let phrase = prompt::request_recovery_phrase();
+            // ask user for a password to secure the wallet
+            let pwd = prompt::create_password();
+            // create wallet
+            let mut w = Wallet::new(phrase)?;
+            w.save_to(WalletFile {
+                path: wallet_path,
+                pwd,
+            })?;
+            w
+        }
+        Command::Interactive => {
+            // load a wallet in interactive mode
+            interactive::load_wallet(&wallet_path)?
+        }
+        _ => {
+            // load wallet from file
+            Wallet::from_file(WalletFile {
+                path: wallet_path,
+                pwd,
+            })?
+        }
+    };
+
+    // attempt to connect wallet
+    #[cfg(windows)]
+    let con = {
+        let t =
+            TransportTCP::with_tcp(&cfg.rusk.rusk_addr, &cfg.rusk.prover_addr)
+                .await;
```

Comment:
`with_tcp` doesn't exist anymore
```suggestion
        let t = TransportTCP::new(&cfg.rusk.rusk_addr, &cfg.rusk.prover_addr);
```